### ISSUE TITLE
False positive on Lint/UselessAssignment

### ISSUE THREAD

**Issue by deepj** [bbatsov/rubocop#4351]
## Expected behavior

Not to be false positive on Lint/UselessAssignment when a variable is proc/lambda with argument(s)

## Actual behavior

The following code is false positive on Lint/UselessAssignment: Useless assignment to variable - `add_bang`. (Lint/UselessAssignment)	

```ruby
  add_bang = ->(value) { "#{value}!" }
  add_bang('bang')
```

I noticed any variable with proc/lambda with argument(s) is evaluated false positive on Lint/UselessAssignment.

## Steps to reproduce the problem

See the code above

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-darwin16)
```

**Comment by Drenmi** [comment]
Technically, if all you want to do is invoke the lambda, you could do:

```ruby
->(value) { "#{value}!" }['bang']
```

**Comment by deepj** [comment]
@Drenmi that's a short reproducible example. I need to use this kind of code for `#map` or other purposes. I guess this is bug than something else :)

**Comment by backus** [comment]
@deepj are you sure it is a bug? In your example you can't call `add_bang` like that. You can do `add_bang.('bang')`, `add_bang.call('bang')`, or `add_bang['bang']`

**Comment by deepj** [comment]
@backus Thanks for correcting me. Just very overworked and mixed up several things into one.
